### PR TITLE
able to hide edit button conditionally

### PR DESCRIPTION
**Pull request by reichie020212**

### CODE REVIEW

**Comment by jeremi** [comment]
@gonzalesedwin1123 @nhatnm0612 Do you know if there is a cleaner way to do this ?

**Comment by nhatnm0612** [comment]
> @gonzalesedwin1123 @nhatnm0612 Do you know if there is a cleaner way to do this ?

why we don't use record rule feature?

**Comment by gonzalesedwin1123** [comment]
> @gonzalesedwin1123 @nhatnm0612 Do you know if there is a cleaner way to do this ?

I think this is a cleaner way to do this:
@reichie020212 you can try this javascript, which is similar to what we use in the 'Create Program' Wizard:
`odoo.define('<<module.custom_form_view>>', function (require) {
"use strict";

var core = require('web.core');
var FormController = require('web.FormController');

var _t = core._t;
var QWeb = core.qweb;

FormController.include({
    updateButtons: function () {
        if (this.modelName === "<<model.name>>") {
            if (this.renderer.state.data.state !== "draft") {
                this.$buttons.find('.o_form_button_edit')
                        .toggleClass('o_hidden', true);
            } else {
                this.$buttons.find('.o_form_button_edit')
                        .toggleClass('o_hidden', false);
            }
        } else {
            this.$buttons.find('.o_form_button_edit')
                    .toggleClass('o_hidden', false);
        }
        var edit_mode = (this.mode === 'edit');
        this.$buttons.find('.o_form_buttons_edit')
            .toggleClass('o_hidden', !edit_mode);
        this.$buttons.find('.o_form_buttons_view')
            .toggleClass('o_hidden', edit_mode);
    },
});

});`

File: spp_change_request/models/change_request.py [hunk 1]
```diff
@@ -152,6 +152,24 @@ def _default_name(self):
         compute="_compute_validation_group_id",
         store=True,
     )
+    hide_button_css = fields.Html(
+        string="CSS", sanitize=False, compute="_compute_hide_button_css", store=False
+    )
```
Comment (gonzalesedwin1123) [inline spp_change_request/models/change_request.py]: Try the javascript solution in my comment.

**Review by gonzalesedwin1123 — CHANGES_REQUESTED** [review]

File: spp_change_request/models/change_request.py (hunk 1 above)
Comment (gonzalesedwin1123) [reply spp_change_request/models/change_request.py]: As @nhatnm0612 suggested, you can try adding a record rule that is applied only to 'edit' for addressing the requirement of allowing the editing of fields only for users assigned to the CR.

File: spp_change_request/models/change_request.py [hunk 2]
```diff
@@ -152,6 +152,24 @@ def _default_name(self):
...
+
+    @api.depends("state")
+    def _compute_hide_button_css(self):
+        for record in self:
+            if (
+                record.state == "draft"
+                and record.create_uid != self.env.user
+                and record.assign_to_id != self.env.user
+            ):
+                # Hide edit button when state is 'draft' and current user is not in either create_uid or assign_to_id
```
Comment (nhatnm0612) [inline spp_change_request/models/change_request.py:167]: @reichie020212 this logic tell me that: if the record state is "draft" && the creator and assignee must be the current user to edit this record?

File: spp_change_request/static/src/js/hide_button.js [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+                    } else {
+                        this.$buttons.find(".o_form_button_edit").toggleClass("o_hidden", false);
+                    }
+                } else {
+                    this.$buttons.find(".o_form_button_edit").toggleClass("o_hidden", false);
+                }
+                var edit_mode = this.mode === "edit";
+                this.$buttons.find(".o_form_buttons_edit").toggleClass("o_hidden", !edit_mode);
+                this.$buttons.find(".o_form_buttons_view").toggleClass("o_hidden", edit_mode);
+            }
```
Comment (nhatnm0612) [inline spp_change_request/static/src/js/hide_button.js:32]:
```suggestion
            var renderStateData = this.renderer.state.data;
            if (!(
                this.$buttons &&
                renderStateData.assign_to_id &&
                renderStateData.assign_to_id.data &&
                renderStateData.create_uid &&
                renderStateData.create_uid.data
            )) {
                return;
            }
            if (
                this.modelName === 'app.change.request' &&
                renderStateData.state === 'draft' &&
                renderStateData.assign_to_id.data.id !== user &&
                renderStateData.create_uid.data.id !== user
            ) {
                this.$buttons.find('.o_form_button_edit').toggleClass('o_hidden', true);
            } else {
                this.$buttons.find('.o_form_button_edit').toggleClass('o_hidden', false);
            }
            var editMode = this.mode === 'edit';
            
             this.$buttons.find(".o_form_buttons_edit").toggleClass("o_hidden", !edit_mode);
             this.$buttons.find(".o_form_buttons_view").toggleClass("o_hidden", edit_mode);
```
really think that guard logic will improve the code readability.

**Review by gonzalesedwin1123 — APPROVED** [review]

**Pull request merged**